dataset/image_extractor/utils/PicselliaImageExtractor.py

```diff
@@ -0,0 +1,95 @@
+import os
+
+import cv2
+from picsellia import Rectangle
+from picsellia.types.enums import TagTarget
+
+from abstract_processor.processor import AbstractProcessor
+
+
+class PicselliaImageExtractor(AbstractProcessor):
+    def __init__(self):
+        super().__init__()
+        context = self.job.sync()["dataset_version_processing_job"]
+
+        self.datalake = self.client.get_datalake(
+            name=context["parameters"].get("datalake_name", "default")
+        )
+        self.dataset_version = self.client.get_dataset_version_by_id(
+            id=context["input_dataset_version_id"]
+        )
+        self.dataset_version_folder = self.dataset_version.version
+
+        self.extracted_dataset_version = self.client.get_dataset_version_by_id(
+            id=context["output_dataset_version_id"]
+        )
+        self.extracted_dataset_version_folder = self.extracted_dataset_version.version
+
+    def _download_ds(self):
+        self.dataset_version.download(target_path=self.dataset_version_folder)
```

Comment:
it can be useful to add a check that the dataset version is object_detection here, otherwise print msg to user that it's only applied to bounding boxes

Reply:
I didn't do it because there's a dataset type constraint on this processing. But for further use of this code, it's better to add it!